insert permissions with backend privileges (RFC #4120) (#4224)
hasura/graphql-engine: docs/graphql/manual/auth/authorization/permission-rules.rst

removes access to it. This preset can be defined for ``insert`` and ``update`` operations. This configuration
is also very useful to avoid sending sensitive user-information in the query and leverage session variables
or static data instead.

.. _backend-only-insert-permissions:

Backend only inserts
^^^^^^^^^^^^^^^^^^^^

If the ``insert`` permission is marked as ``backend_only: true``, the mutation is accessible to the

> given role only if the ``x-hasura-backend-privilege`` session variable exists and set to ``true``.

Reviewer: ```suggestion
given role only if the ``x-hasura-backend-privilege`` session variable exists and is set to ``true``.
```

removes access to it. This preset can be defined for ``insert`` and ``update`` operations. This configuration
is also very useful to avoid sending sensitive user-information in the query and leverage session variables
or static data instead.

.. _admin-only-insert-permissions:

> Admin Only Inserts

Reviewer: ```suggestion
Admin only inserts
```

^^^^^^^^^^^^^^^^^^

> If the ``insert`` permission is marked as ``admin_only: true`` then the mutation is accessbile to

Reviewer: ```suggestion
If the ``insert`` permission is marked as ``admin_only: true``, the mutation is accessible to the
```

> given role (``x-hasura-role`` header) only if admin secret (``x-hasura-admin-secret`` header) is sent in the request.

Reviewer: ```suggestion
given role (``x-hasura-role`` header) only if the admin secret (``x-hasura-admin-secret`` header) is sent in the request.
```

> If the admin secret is not configured then the mutation is open for the given role.

Reviewer: ```suggestion
If the admin secret is not configured, the mutation is open for the given role.
```
